k8s.go

```diff
@@ -0,0 +1,247 @@
+package profitbricks
+
+import "net/http"
+
+type KubernetesClusters struct {
+	// URL to the collection representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// Unique representation for Kubernetes Cluster as a collection on a resource.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// Array of items in that collection
+	// Read Only: true
+	Items []*KubernetesCluster `json:"items"`
+
+	// The type of resource within a collection
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+}
+
+// KubernetesCluster kubernetes cluster
+type KubernetesCluster struct {
+	// URL to the object representation (absolute path)
+	// Read Only: true
+	// Format: uri
+	Href string `json:"href,omitempty"`
+
+	// The resource's unique identifier.
+	// Read Only: true
+	ID string `json:"id,omitempty"`
+
+	// metadata
+	Metadata *NoStateMetaData `json:"metadata,omitempty"`
+
+	// properties
+	// Required: true
+	Properties *KubernetesClusterProperties `json:"properties"`
+
+	// The type of object
+	// Read Only: true
+	// Enum: [k8s]
+	PBType string `json:"type,omitempty"`
+
+	// Entities of a cluster
+	Entities KubernetesClusterEntities `json:"entities"`
+}
+
+type KubernetesClusterEntities struct {
+	// NodePools of a cluster
+	NodePools *KubernetesNodePools `json:"nodepools,omitempty"`
+}
+
+// KubernetesClusterProperties kubernetes cluster properties
+type KubernetesClusterProperties struct {
+	// A Kubernetes Cluster Name. Valid Kubernetes Cluster name must be 63 characters or less and must be empty
```

Comment:
As the field is required I am surprised that "empty" should be valid. Have you tried?

Reply:
Empty is not valid, changed comment